bart/sched/pelt.py

```diff
@@ -0,0 +1,377 @@
+#    Copyright 2015-2016 ARM Limited
+#    Copyright 2016 Google Inc. All Rights Reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+PELT Simulation Model
+"""
+
+from collections import namedtuple as _namedtuple
+from pandas import DataFrame as _DataFrame
+
+################################################################################
+# Task Model
+################################################################################
+
+class Task(object):
+
+    pelt_sample_us = 1024
+    pelt_max = 1024
+
+    def __init__(self, period_ps, run_ps, start_ps=0,
```

Comment:
BTW up to you if you think it's worth validating but there's a divison by zero if period is 0. Also if `run_ps` > `period_ps` we get nonsense util_avg. Both of these are invalid parameters but might be worth `raise`ing to make those errors more obvious to users?